plugins/languages/treesitter/ts-autotag.nix

```diff
@@ -1,78 +1,102 @@
 {
-  pkgs,
   lib,
   helpers,
   config,
+  pkgs,
   ...
 }:
 with lib;
-{
-  options.plugins.ts-autotag = helpers.neovim-plugin.extraOptionsOptions // {
-    enable = mkEnableOption "nvim-ts-autotag";
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "ts-autotag";
+  originalName = "nvim-ts-autotag";
+  luaName = "nvim-ts-autotag";
+  defaultPackage = pkgs.vimPlugins.nvim-ts-autotag;
 
-    package = helpers.mkPluginPackageOption "ts-autotag" pkgs.vimPlugins.nvim-ts-autotag;
+  maintainers = [ maintainers.GaetanLepage ];
 
-    filetypes = helpers.defaultNullOpts.mkNullable (with types; listOf str) ''
+  # TODO introduced 2024-06-17: remove 2024-08-17
+  deprecateExtraOptions = true;
+  imports =
+    map
+      (
+        optionName:
+        mkRemovedOptionModule
+          [
+            "plugins"
+            "ts-autotag"
+            optionName
```

Comment:
I expect removing top-level options is a common pattern, maybe mkNeovimPlugin should have a `removedOptions` attrset arg? :thinking:

Reply:
Well. It could. But the description for each one is often specific, so we could only factor out so much.

Reply:
I meant `removedOptions` as an `{ name = description; }`, but yes, that'd only be useful for top-level